processor/internal/destination_transformer/destination_transformer.go

```diff
@@ -0,0 +1,310 @@
+package destination_transformer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+
+	transformerclient "github.com/rudderlabs/rudder-server/internal/transformer-client"
+	"github.com/rudderlabs/rudder-server/processor/integrations"
+	"github.com/rudderlabs/rudder-server/processor/internal/transformer_utils"
+	"github.com/rudderlabs/rudder-server/processor/types"
+	"github.com/rudderlabs/rudder-server/utils/httputil"
+	reportingTypes "github.com/rudderlabs/rudder-server/utils/types"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+type DestTransformer struct {
+	config struct {
+		destTransformationURL   string
+		maxRetry                config.ValueLoader[int]
+		failOnError             config.ValueLoader[bool]
+		maxConcurrency          int
+		maxRetryBackoffInterval config.ValueLoader[time.Duration]
+		timeoutDuration         time.Duration
+	}
+	guardConcurrency chan struct{}
+	conf             *config.Config
+	log              logger.Logger
+	stat             stats.Stats
+	client           transformerclient.Client
+}
+
+type Opt func(*DestTransformer)
+
+func WithClient(client transformerclient.Client) Opt {
+	return func(s *DestTransformer) {
+		s.client = client
+	}
+}
+
+func (d *DestTransformer) SendRequest(ctx context.Context, clientEvents []types.TransformerEvent, batchSize int) types.Response {
```

Comment:
Why don't we keep the `batchSize` reloadable configuration variable inside the transformer client itself? We now have different clients for each type of transformation.